### ISSUE TITLE
Custom Layers for recursive interation bitween layers

### ISSUE THREAD

**Issue by tiagocalumbi** [fchollet/keras#5487]
I have a question. I'm trying to make a CNN, the return of model feeds first Dence layer. Can I help?  
e.g:

```
model = Sequential()

model.add(Convolution2D(32, 3, 3, border_mode='same',
                        input_shape=X_train.shape[1:]))
model.add(Activation('relu'))
model.add(Convolution2D(32, 3, 3))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Convolution2D(64, 3, 3, border_mode='same'))
model.add(Activation('relu'))
model.add(Convolution2D(64, 3, 3))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(nb_classes))
model.add(Activation('softmax'))

```
I Should have the output value feeds layer Dense (512)
 1. Dense (512 + 10, name='d1') output  d2 with input here.
 2. Activation()  
 3. Dropout(522)  
 4. Dense (10, name='d2')

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by unrealwill** [comment]
Hello,
You can't have a cyclic graph in keras.

What you can nether the less do is : first obtain a "class" prediction like you are doing, Then concatenate to the result of the cnn. And redo an other prediction. You will probably need to use the functional API. You may also be able to reuse weights (if for the first prediction you do it like the second but concatenate a zero vector instead of the prediction you do not yet have).

```
cnn = Flatten(...)
zero = ZeroGeneratingLayerOfCorrectBatchShape(cnn) #you 'll need to write this custom layer on you own :)
d1 = Dense(512,activation="relu") #the layers weights will be reused multiple times
d2 = Dense(10,activation="softmax") #the layers weights will be reused multiple times

merge =Merge( mode="concat") ([cnn,zero])
for i in range( nbIter ):
  out = d2(d1(merge))
  merge = Merge( mode="concat") ([cnn,out])
return out
```

What we just did can also be implemented using a recursive layer. (Not exactly in keras because you will have to implement your own custom recurrent layer if you want to feed the final output in).

This will approach the prediction by iterating : 
```
cnn = Flatten(..)
rep = Repeat( nbIter )(cnn)
out = SimpleRNNOrGRUOrLSTM( 512,return_sequence=false )(rep)
```

One last alternative is to have a model which take as input of the model the first prediction (or an estimate). Then you make a new prediction using this estimate and it gives you an other value (which you can plug in again until convergence if you want).

**Comment by tiagocalumbi** [comment]
@unrealwill Think a lot. I'm sorry but I don't know custom Layer for ZeroGeneratingLayerOfCorrectBatchShape. You Could help me? 

Before, I was trying to make with sequencial model and Merge alone.  But I saw this was difficult.

**Comment by unrealwill** [comment]
By ZeroGeneratingLayerOfCorrectBatchShape I mean a layer which outputs a zero tensor of shape (batch_size,outputdim).

It is a pretty straightforward custom layer which is a good exercise to improve your keras knowledge.
https://keras.io/layers/writing-your-own-keras-layers/

Extra hints you can use K.sum( axis=1), K.repeat_elements(axis=1,self.outputdim) and  K.zeros_like

**Comment by tiagocalumbi** [comment]
@unrealwill I'm sorry again. I still don't found. Could you help me?  :) ahah
